cryptoid/src/adapter.ts

```diff
@@ -0,0 +1,47 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { util } from '@chainlink/ea-bootstrap'
+
+const customParams = {
+  blockchain: ['blockchain', 'coin'],
+  endpoint: false,
+}
+
+const endpointToApiParam: { [key: string]: string } = {
+  difficulty: 'getdifficulty',
+  height: 'getblockcount',
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+  const blockchain = validator.validated.data.blockchain.toLowerCase()
+
+  const key = util.getRandomRequiredEnv('API_KEY')
+  const apiParam = endpointToApiParam[endpoint]
```

Comment:
Sorry for being picky here, as I understand `q` param is the function you want to call from cryptoId, so something like `cryptoIdFunction` or `apiFunction` would be better.

Reply:
`q` is the query parameter of the cryptoid endpoint. Not sure if it should be called function. In essence though, it has identical name as the JSON RPC methods.

I was thinking of something to let us know that the endpoint is mapped to the actual api's parameter.

Reply:
Changed to `apiFunctionName` after discussion.